rfcs/0030-review-workflow.md

```diff
@@ -0,0 +1,147 @@
+
+## Tooling
+
+This workflow is not currently possible due to GitHub's restrictions. Namely
+only people with commit access can set tags on a PR. This can be resolved
+through tooling. Work on this is [in progress](https://github.com/NixOS/ofborg/pull/216).
+
+The tooling should then initialize the PR status on new PRs. In general this
+is going to be `needs:review`. One exception of this are PRs that are work in
+progress (indicated by a `WIP` tag in the title). Those should start at
+`needs:work`. Other exceptions may include RFC type PRs, which may warrant their
+own state (`needs:opinions`?).
+
+To reduce friction for new contributors, the tooling could later be expaned
+to explain this process to first time contributors.
+
+## Advantages
+
+All the above issues should be resolved by this:
+
+1. There is no reason anymore to preferably review recent PRs. Before, recent
+   commit were the only ones where one can be sure that they are not reviewed
+   yet (and thus in a `needs:work` state). Now, reviewers can randomly choose
+   and `needs:review` PR.
+
+2. Similarly, the reviewer may change at any time. If the first reviewer forgets
+   or loses interest in a PR, the PR will just end up in the `needs:review` list
+   again and can easily find a new reviewer.
+
+3. People without commit access can help reviewing commits and lessen the load
+   on committers without putting the reviewed PRs at a disadvantage.
+
+4. We can easily gather statistics. In the long run, we might even implement
+   some sort of automated triage where we regularly ping `needs:work` PRs and
+   close them after the authors went unresponsive for a while. Or we could
+   tag them as `needs:takeover`. But this is out of scope.
+
+
+An example PR might go like this. `contributor` creates the PR, `reviewer1`
+reviews without commit access, `reviewer2` reviews with commit access.
+
+> contributor: hello: 42.0 -> 43.0 [PR text]
+
+[the bot automatically sets the new PR to `needs:review`]
+
+> reviewer1: Looks good to me. Just one little nitpick: Please add a waving emoji to the name.
+> @GrahamcOfBorg status needs:work
+
+> contributor: Good point, more emojis are always good. Done.
+> @GrahamcOfBorg status needs:review
+
+> reviewer1: Thanks! I think this is ready for merge.
+> @GrahamcOfBorg status needs:merge
+
+> reviewer2: LGTM
+> @GrahamcOfBorg build-and-merge
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Contributing would be a bit more complicated. There is one more bit of
+  knowledge new contributors need. It is possible that some contributors forget
+  to set the PR to `needs:review` again after a fixup. That might be resolved
+  by some form of automatic triage as explained above. It would also be not much
+  than the current status, since the first reviewer will still be notified by
+  GitHub.
+
+- We rely even more on the tooling than we do now. However the general workflow
+  should be portable to different tooling and might even work better there.
```

Comment:
Clearification needed: Is this (more tooling) actually a drawback? I don't think so, acutally! As long as it works and is properly documented, I don't mind only interacting with a bot from first patch to successfull merge!

Reply:
I agree this is a drawback because with a different platform such as e.g GitLab, the toolings will need a rewrite / adjustments to the new APIs etc.